build-artifacts fails with Error: invalid semantic version

### What happened?

We're seeing the CI step for build-artifacts failing at least on main, unclear if release branches are affected yet.

Example: https://github.com/crossplane/crossplane/actions/runs/26253873188/job/77271667730

```
building '/nix/store/xin74bhshsj7864viskk37v224vl95ry-crossplane-helm-chart-pis-v2.4.0-rc.0.17.g68420bc9a.drv'...
crossplane-helm-chart-pis-v2.4.0-rc.0.17.g68420bc9a> Error: invalid semantic version
error: Cannot build '/nix/store/xin74bhshsj7864viskk37v224vl95ry-crossplane-helm-chart-pis-v2.4.0-rc.0.17.g68420bc9a.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/i6si24rx9jaf7q962kzx9iisb5n9m76c-crossplane-helm-chart-pis-v2.4.0-rc.0.17.g68420bc9a
       Last 1 log lines:
       > Error: invalid semantic version
       For full logs, run:
         nix log /nix/store/xin74bhshsj7864viskk37v224vl95ry-crossplane-helm-chart-pis-v2.4.0-rc.0.17.g68420bc9a.drv
error: Cannot build '/nix/store/1rqfp3h80z10a1zws8bkkm0xpzk2j4x7-crossplane-release-apis-v2.4.0-rc.0.17.g68420bc9a.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/jvb1xmrbbjzq2zdkhd4b4b6790fj4ppv-crossplane-release-apis-v2.4.0-rc.0.17.g68420bc9a
error: Build failed due to failed dependency
Error: Process completed with exit code 1.
```

This is very suspiciously showing up after we added new git tags today during the v2.3 release for the apis module, e.g.:
```
❯ git tag --list "apis/*"
apis/v2.3.0
apis/v2.3.0-rc.1
apis/v2.4.0-rc.0
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -242,7 +242,7 @@ jobs:
       # tracked files, so we set it in flake.nix before building.
       - name: Set Version
         run: |
-          VERSION=$(git describe --dirty --always --tags | sed 's/-/./2g')
+          VERSION=$(git describe --dirty --always --tags --match 'v[0-9]*' | sed 's/-/./2g')
           echo "VERSION=$VERSION" >> "$GITHUB_ENV"
           sed -i "s|buildVersion = null;|buildVersion = \"$VERSION\";|" flake.nix
 

